ticker/src/main/java/com/robinhood/ticker/TickerCharacterList.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2016 Robinhood Markets, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.robinhood.ticker;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This is the primary class that Ticker uses to determine how to animate from one character
+ * to another. The provided string dictates what characters will appear between
+ * the start and end characters.
+ *
+ * <p>For example, given the string "abcde", if the view wants to animate from 'd' to 'b',
+ * it will know that it has to go from 'd' to 'c' to 'b', and these are the characters
+ * that show up during the animation scroll.
+ *
+ * @author Jin Cao, Robinhood
+ */
+public class TickerCharacterList {
+    private final int numOriginalCharacters;
+    // The saved character list will always be of the format: EMPTY, list, list
+    private final char[] characterList;
+    // A minor optimization so that we can cache the indices of each character.
+    private final Map<Character, Integer> characterIndicesMap;
+
+    public TickerCharacterList(String characterList) {
+        if (characterList.contains(Character.toString(TickerUtils.EMPTY_CHAR))) {
+            throw new IllegalArgumentException(
+                    "You don't need to include TickerUtils.EMPTY_CHAR in the character list.");
```

Comment:
message is a bit misleading - "You cannot include TickerUtils.EMPTY_CHAR in the character list"